### PR TITLE
chore(dependencies): Increase opencv-python-headless upper bound

### PR DESCRIPTION
**Pull request by traversaro**
## Title

chore(dependencies): Increase opencv-python-headless upper bound

## Type / Scope

- **Type**:  Chore

## Summary / Motivation

- OpenCV 4.13 is available, and in particular in https://github.com/huggingface/lerobot/pull/3114 opencv-python was updated to 4.13, keeping opencv-python-headless to an earlier version is error prone

## What changed

- The upper bound of `opencv-python-headless` in `pyproject.toml` is raised from `<4.13.0` to `<4.14.0`.

## How was this tested (or how to run locally)

- Test locally run

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [x] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -76,7 +76,7 @@ dependencies = [
     "torchvision>=0.21.0,<0.26.0",
 
     "einops>=0.8.0,<0.9.0",
-    "opencv-python-headless>=4.9.0,<4.13.0",
+    "opencv-python-headless>=4.9.0,<4.14.0",
```
[the remark this answers is not in the document]
Comment (traversaro) [reply pyproject.toml:79]: Solved.

[the remark this answers is not in the document]
Comment (traversaro) [reply pyproject.toml:79]: @imstevenpmwork do you have any opinion on this? I checked other PRs like https://github.com/huggingface/lerobot/pull/2897 and it seems that `requirements-*.txt` files are typically not updated in PRs that modify `pyproject.toml`.

**Comment by imstevenpmwork** [comment]
Hey @traversaro 

I'm happy with bumping the dependency if it has been tested already. But I'm afraid I didn't understand your explanation, currently our ceil is `<4.13.0` and in the `requirements-*` is indeed `4.12.*`, so no `4.13`.

For clarification, the motive behind the `requirements-*` is just for future reproducibility (mostly for maintainers), for us to be able to come back to a specific `minor` release and check what were the dependency tree back then. Currently we don't do this for `patch` releases. If you depend on `lerobot` I suggest ignoring these files unless you really want to pin all the dependencies at the application layer.

**Comment by traversaro** [comment]
> I'm happy with bumping the dependency if it has been tested already.

Yes, I tested locally and the opencv-related test work fine.

> But I'm afraid I didn't understand your explanation, currently our ceil is <4.13.0 and in the requirements-* is indeed 4.12.*, so no 4.13.

Yes, that was confusing on my side. The real reason why we want to be able to install `lerobot` with opencv==4.13.0 is that conda-forge recently closed it `libopencv==4.13.0` migration (see https://conda-forge.org/status/migration/?name=libopencv4130) so all the latest packages in conda-forge (and robostack) will be built with opencv 4.13, and we would like to be able to install those with lerobot.

The reason why I brought up the `requirements-*.txt`  file, is that due to bound on `opencv-python-headless` , if in a clean environement I install lerobot[all], this happens:

~~~
conda create -n testlerobot -c conda-forge python=3.12.* egl-probe uv ffmpeg
conda activate testlerobot 
uv pip install lerobot[all]
uv pip list | grep opencv
Using Python 3.12.13 environment at: .pixi/envs/conda/envs/testlerobot3
opencv-python             4.13.0.92
opencv-python-headless    4.12.0.88
~~~

Basically, both `opencv-python` and `opencv-python-headless` are installed, but at different versions. This is exactly what is also happening in the `requirements-*.txt` files, and an additional reason (but not the primary one, for me) to raise the opencv upper bound.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (traversaro) [reply pyproject.toml:79]: As mentionen in https://github.com/huggingface/lerobot/pull/3120#issuecomment-4025571579, requirements-*.txt files are only updated on releases, so I think we can ignore copilot comments.
